internal/impl/questdb/output.go

```diff
@@ -0,0 +1,526 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package questdb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	qdb "github.com/questdb/go-questdb-client/v3"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+func questdbOutputConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Summary("Pushes messages to a QuestDB table").
+		Description("Important: We recommend that the dedupe feature is enabled on the QuestDB server. "+
+			"Please visit https://questdb.io/docs/ for more information about deploying, configuring, and using QuestDB."+
+			service.OutputPerformanceDocs(true, true)).
+		Categories("Services").
+		Fields(
+			service.NewOutputMaxInFlightField(),
+			service.NewBatchPolicyField("batching"),
+			service.NewStringField("address").
+				Description("Address of the QuestDB server's HTTP port (excluding protocol)").
+				Example("localhost:9000"),
+			service.NewStringField("username").
+				Description("Username for HTTP basic auth").
+				Optional().
+				Secret(),
+			service.NewStringField("password").
+				Description("Password for HTTP basic auth").
+				Optional().
+				Secret(),
+			service.NewStringField("token").
+				Description("Bearer token for HTTP auth (takes precedence over basic auth username & password)").
+				Optional().
+				Secret(),
+			service.NewBoolField("tlsEnabled").
+				Description("Use TLS to secure the connection to the server").
+				Optional(),
+			service.NewStringField("tlsVerify").
+				Description("Whether to verify the server's certificate. This should only be used for testing as a "+
+					"last resort and never used in production as it makes the connection vulnerable to "+
+					"man-in-the-middle attacks. Options are 'on' or 'unsafe_off'.").
+				Optional().
+				Default("on").
+				LintRule(`root = if ["on","unsafe_off"].contains(this != true { ["valid options are \"on\" or \"unsafe_off\"" ] }`),
```

Comment:
We have a service.NewTLSToggledField("tls") that has default fields supported everywhere - can we please us that?

https://github.com/redpanda-data/connect/blob/1c664da8ab51b796bb5e11189fdd41f2b41d9125/internal/impl/kafka/input_kafka_franz.go#L123

Reply:
Yep, how's this? Our sender uses the tlsConfig contained in the unmarshalled conf.